src/groupeddataframe/splitapplycombine.jl

```diff
@@ -780,14 +805,36 @@ end
 Aggregate(f) = Aggregate(f, nothing)
 
 check_aggregate(::typeof(var)) = Aggregate(var)
+validate_aggregate(::typeof(var), ::AbstractVector{<:Union{Missing, Number}}) = true
+
 check_aggregate(::typeof(var∘skipmissing)) = Aggregate(var, !ismissing)
+validate_aggregate(::typeof(var∘skipmissing), ::AbstractVector{<:Union{Missing, Number}}) = true
+
 check_aggregate(::typeof(std)) = Aggregate(std)
+validate_aggregate(::typeof(std), ::AbstractVector{<:Union{Missing, Number}}) = true
+
 check_aggregate(::typeof(std∘skipmissing)) = Aggregate(std, !ismissing)
+validate_aggregate(::typeof(std∘skipmissing), ::AbstractVector{<:Union{Missing, Number}}) = true
+
 check_aggregate(::typeof(first)) = Aggregate(first)
+validate_aggregate(::typeof(first), v::AbstractVector) = eltype(v) === Any ? false : true
+validate_aggregate(::typeof(first), ::AbstractVector{<:Union{Missing, MULTI_COLS_TYPE, AbstractVector}}) = false
```

Comment:
We indeed can discuss this and this relates to what @pdeffebach wants that we allow destruction into multiple columns.

Just to be clear the `MULTI_COLS_TYPE` part is not really problematic in my opinion - it just makes sure we throw an error (as we disallow it now), so in the future we can process it correctly. This is what we try to do consistently everywhere, so that post 1.0 changes here will be non-breaking, as they will currently throw an error (this is your pet trick to handle SemVer AFAICT :smile:).

A more problematic case is `AbstractVector` as it was simply inconsistent between slow and fast paths (because fast path was not expanding it and slow path does pseudo-broadcasting).

So there actually two questions:
* if we want some `MultiCol` wrapper in the future or just allow multiple cols to be returned and silently accept it (I thought @pdeffebach wanted to silently accept this)
* do we require to opt-in for pseudo-broadcasting, we never required it in the past, and it was the default, I think we should keep it.

Although it is a legacy from the very distant past. If I were designing it now I would never expand anything neither in rows nor in columns, just store one result per group in `combine` and then say to users to use `flatten` to flatten it. But this is a completely different design in comparison to what we have now, so this is just a side comment.

Reply:
Actually I had forgotten that we don't allow returning `MULTI_COLS_TYPE` currently. So at least that's safe.

`AbstractVector` remains problematic though. I wonder whether there are strong use cases for pseudo-broadcasting. It would be safer to make this opt-in for 1.0, otherwise we don't allow working with data frames whose cells contain vectors.

(BTW, instead for `MultiCol`, maybe we could reuse `AsTable`, but to wrap the returned value this time.)

Reply:
I have thought about it when cleaning up the rules of pseudo-broadcasting recently. Here is what we have on 0.21:
```
julia> df = DataFrame(g=[1,2,3], x=[1:3, 4:6, 7:9])
3×2 DataFrame
│ Row │ g     │ x        │
│     │ Int64 │ UnitRan… │
├─────┼───────┼──────────┤
│ 1   │ 1     │ 1:3      │
│ 2   │ 2     │ 4:6      │
│ 3   │ 3     │ 7:9      │

julia> gdf = groupby(df, :g)
GroupedDataFrame with 3 groups based on key: g
First Group (1 row): g = 1
│ Row │ g     │ x        │
│     │ Int64 │ UnitRan… │
├─────┼───────┼──────────┤
│ 1   │ 1     │ 1:3      │
⋮
Last Group (1 row): g = 3
│ Row │ g     │ x        │
│     │ Int64 │ UnitRan… │
├─────┼───────┼──────────┤
│ 1   │ 3     │ 7:9      │

julia> combine(gdf, :x => first) # this is a wrong result and will be fixed by this PR to match what we have below
3×2 DataFrame
│ Row │ g     │ x_first  │
│     │ Int64 │ UnitRan… │
├─────┼───────┼──────────┤
│ 1   │ 1     │ 1:3      │
│ 2   │ 2     │ 4:6      │
│ 3   │ 3     │ 7:9      │

julia> combine(gdf, :x => x -> first(x)) # this is the intended output
9×2 DataFrame
│ Row │ g     │ x_function │
│     │ Int64 │ Int64      │
├─────┼───────┼────────────┤
│ 1   │ 1     │ 1          │
│ 2   │ 1     │ 2          │
│ 3   │ 1     │ 3          │
│ 4   │ 2     │ 4          │
│ 5   │ 2     │ 5          │
│ 6   │ 2     │ 6          │
│ 7   │ 3     │ 7          │
│ 8   │ 3     │ 8          │
│ 9   │ 3     │ 9          │

julia> combine(gdf, :x => Ref∘first) # this is a currently intended method to protect from unwrapping - just like in standard broadcasting
3×2 DataFrame
│ Row │ g     │ x_function │
│     │ Int64 │ UnitRange… │
├─────┼───────┼────────────┤
│ 1   │ 1     │ 1:3        │
│ 2   │ 2     │ 4:6        │
│ 3   │ 3     │ 7:9        │
```
So in short - we allow working with data frames that contain vectors as:
1. normally vectors of vectors will be returned and it is not a problem
2. if user unwraps the vector (like above - with `first`) then `Ref` can be used as in Base to protect the result

Reply:
Let's continue this discussion in a separate issue?

Reply:
OK - can you please open the issue explaining what you would want to change? (given the three key considerations: 1) currently we unwrap vectors, 2) `Ref` protects, 3) in the future we want to add support for multiple columns passing)